manifests/install.pp

```diff
@@ -69,18 +69,26 @@
   }
 
   $installDir = "${java::params::installRootDir}/jdk-${vendor}-${version}-${arch}"
+  
+  $major = inline_template('<%= scope.lookupvar(\'version\').split(\'-\')[0].gsub(\'.\', \'_\') %>')
+  # convert $major value to an integer
+  $majora = scanf($major, "%i") 
+  $majori = $majora[0]
 
   case $vendor {
     # Java <= 6 are using the vendor sun
     /(sun)/ : {
       # Extract the major version removing the beta
-      $major   = inline_template('<%= scope.lookupvar(\'version\').split(\'-\')[0].gsub(\'.\', \'_\') %>')
       $file    = "jdk-${major}-linux-${arch}.bin"
       $jdk_dir = "jdk-${major}-${vendor}-${arch}"
     }
     # Java > 6 are using the vendor oracle
     /(oracle)/ : {
-      $file    = "jdk-${version}-linux-${arch}.tar.gz"
+      if ( $majori < 9 ) {
+        $file    = "jdk-${version}-linux-${arch}.tar.gz"
+      } else {
```

Comment:
we don't really know if `$majori` is an integer
don't you think we should test the case where `$majori > 9` then have an else which `fail(...)` ?

Reply:
$majori is now an integer for sure as a validate_integer is added.

As it's already a if..elase, I don't get your question 
> don't you think we should test the case where $majori > 9 then have an else which fail(...) ?